src/menu.rs

```diff
@@ -490,60 +489,66 @@ impl<T> RawMenuItem<T> {
         properties
     }
 
-    pub(crate) fn diff(
-        &self,
-        other: Self,
-    ) -> Option<(HashMap<String, OwnedValue>, Vec<String>)> {
+    pub(crate) fn diff(&self, other: Self) -> Option<(HashMap<String, OwnedValue>, Vec<String>)> {
         let default = Self::default();
         let mut updated_props: HashMap<String, OwnedValue> = HashMap::new();
         let mut removed_props = Vec::new();
         if self.r#type != other.r#type {
             if other.r#type == default.r#type {
                 removed_props.push("type".into());
             } else {
-                updated_props.insert("type".into(), Value::from(other.r#type.to_string()).into());
+                updated_props.insert("type".into(), OwnedValue::try_from(other.r#type).unwrap());
```

Comment:
Where is the `impl TryFrom<ItemType> for OwnedValue`?

The `to_string` is not just a type convert, are you sure the `Value` is a `Value::Str`?